libbeat/processors/fingerprint/fingerprint.go

```diff
@@ -49,21 +48,23 @@ func New(cfg *common.Config) (processors.Processor, error) {
 		return nil, makeErrConfigUnpack(err)
 	}
 
-	fields := common.MakeStringSet(config.Fields...)
+	// The fields array must be sorted, to guarantee that we always
+	// get the same hash for a similar set of configured keys.
+	// The call `ToSlice` always returns a sorted slice.
+	fields := common.MakeStringSet(config.Fields...).ToSlice()
```

Comment:
Out of curiosity, is there a difference in calling `.ToSlice()` here vs. how it was before, a few lines below? Either way the value of `p.fields` would be sorted, no? I'm just wondering if you are making this change to try and do all the operations on `config.Fields` in one place (this line) or if there's some other more fundamental reason that could prevent a bug or something else.

Reply:
I just moved it here in order to 'cleanly' add the code comment. Operationally it makes no difference. It's not obvious that `ToSlice` already sorts and not immediately obvious that we always need the keys to be sorted, so we should document this (at first I thought we have a bug thanks to StringSet storing keys in randomized order).

Often I prefer to have fully 'initialized' variables that I don't need to modify or execute other operations on when constructing another object. This gives each code block a self-contained purpose (almost like a function).